designer/client/__tests__/FieldEdit.jest.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useReducer } from "react";
+import { render } from "@testing-library/react";
+import { Data } from "@xgovformbuilder/model";
+import { DataContext } from "../context";
+import {
+  ComponentContext,
+  componentReducer,
+  initComponentState,
+} from "../reducers/component/componentReducer";
+import { FieldEdit } from "../field-edit";
+import { initI18n } from "../i18n";
+
+describe("Field Edit", () => {
+  beforeEach(() => {
+    initI18n();
+  });
+
+  let data = new Data({
+    pages: [
+      {
+        title: "First page",
+        path: "/first-page",
+        components: [
+          {
+            name: "IDDQl4",
+            title: "abc",
+            schema: {},
+            options: {
+              required: true,
+            },
+            type: "RadiosField",
+            values: {
+              type: "static",
+              items: [
+                { label: "My item", value: "12", children: [] },
+                {
+                  label: "Item 2",
+                  hint: "My hint",
+                  value: "11",
+                  condition: "Abcewdad",
+                  children: [],
+                },
+                {
+                  label: "Item 3",
+                  value: "13",
+                  children: [{ type: "TextField" }],
+                },
+              ],
+            },
+          },
+        ],
+      },
+    ],
+    lists: [
+      {
+        name: "myList",
+        title: "My list",
+        type: "number",
+        items: [{ text: "An item", description: "A hint", value: "12" }],
+      },
+      {
+        name: "myOtherList",
+        title: "",
+        type: "string",
+        items: [{ text: "An item", description: "A hint", value: "12" }],
+      },
+    ],
+  });
+  const dataValue = { data, save: jest.fn() };
+
+  const TestComponentContextProvider = ({
+    children,
+    dataValue,
+    componentValue,
+  }) => {
+    const initComponentValue = (initialState: any) => {
+      return !!componentValue ? componentValue : initialState;
```

Comment:
you can just do this with nullish coalescing - fine as it is though
```suggestion
        return componentValue ?? initialState; //if componentValue is undefined or null (nullish), default to initialState
```